agent/engine/docker_task_engine.go

```diff
@@ -443,6 +443,8 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 	}
 
 	hostConfig, hcerr := task.DockerHostConfig(container, containerMap)
+	hostConfig.NetworkMode = "host"
```

Comment:
Looks like the changes in `agent/api/task.go` should have set this already, is it a case where it gets overridden by parsing the docker config JSON string?

It looks like `tas.DockerHostConfig` is what produces the host config object in all cases, I feel like we should only have to make changes to that function to get it working.